src/event.rs

```diff
@@ -0,0 +1,530 @@
+use crate::{
+	hex_utils, ChannelManager, Error, FilesystemPersister, LdkLiteChainAccess, LdkLiteConfig,
+	NetworkGraph, PaymentInfo, PaymentInfoStorage, PaymentStatus,
+};
+
+#[allow(unused_imports)]
+use crate::logger::{
+	log_error, log_given_level, log_info, log_internal, log_trace, log_warn, FilesystemLogger,
+	Logger,
+};
+
+use lightning::chain::chaininterface::{BroadcasterInterface, ConfirmationTarget, FeeEstimator};
+use lightning::chain::keysinterface::KeysManager;
+use lightning::ln::PaymentHash;
+use lightning::routing::gossip::NodeId;
+use lightning::util::events as ldk_events;
+use lightning::util::persist::KVStorePersister;
+use lightning::util::ser::{MaybeReadable, Readable, ReadableArgs, Writeable, Writer};
+
+use bitcoin::secp256k1::Secp256k1;
+use rand::{thread_rng, Rng};
+use std::collections::{hash_map, VecDeque};
+use std::sync::{Arc, Condvar, Mutex};
+use std::thread;
+use std::time::Duration;
+
+/// The event queue will be persisted under this key.
+pub(crate) const EVENTS_PERSISTENCE_KEY: &str = "events";
+
+/// An event emitted by [`LdkLite`] that should be handled by the user. This will more often than
+/// not wrap a LDK event accessible via the `inner` field.
+#[derive(Debug, Clone)]
+pub enum LdkLiteEvent {
+	/// A payment we sent was successful.
+	PaymentSuccessful {
+		/// The hash of the payment.
+		payment_hash: PaymentHash,
+		/// The wrapped LDK event.
+		inner: ldk_events::Event,
+	},
+	/// A payment we sent has failed.
+	PaymentFailed {
+		/// The hash of the payment.
+		payment_hash: PaymentHash,
+		/// The wrapped LDK event.
+		inner: ldk_events::Event,
+	},
+	/// A payment has been received.
+	PaymentReceived {
+		/// The hash of the payment.
+		payment_hash: PaymentHash,
+		/// The value, in thousandths of a satoshi that has been received.
+		amount_msat: u64,
+		/// The wrapped LDK event.
+		inner: ldk_events::Event,
+	},
+	// TODO: Implement after a corresponding LDK event is added.
+	//ChannelOpened {
+	//},
+	/// A channel has been closed.
+	ChannelClosed {
+		/// The channel_id of the channel which has been closed.
+		channel_id: [u8; 32],
+		/// The wrapped LDK event.
+		inner: ldk_events::Event,
+	},
+	// TODO: Implement on-chain events when better integrating with BDK wallet sync.
+	//OnChainPaymentSent {
+	//},
+	//OnChainPaymentReceived {
+	//}
+}
+
+impl Readable for LdkLiteEvent {
```

Comment:
Yeah, this is up for debate. Right now it's not feasible to do this since the serialization macros are not exported by LDK, and it seem up in the air if they will be (see the discussion in https://github.com/lightningdevkit/rust-lightning/pull/1551).

Reply:
Alright, I now followed-up on above PR and will switch to the macros once they are exported upstream. Opened https://github.com/lightningdevkit/ldk-lite/issues/23 to track this.